test/common/upstream/load_balancer_impl_test.cc

```diff
@@ -1156,6 +1156,42 @@ TEST_P(RoundRobinLoadBalancerTest, WeightedSeed) {
   EXPECT_EQ(hostSet().healthy_hosts_[1], lb_->chooseHost(nullptr));
 }
 
+// Validate that low weighted hosts will be chosen when the LB is created.
+TEST_P(RoundRobinLoadBalancerTest, WeightedInitializationPicksAllHosts) {
+  TestScopedRuntime scoped_runtime;
+  scoped_runtime.mergeValues({{"envoy.reloadable_features.locality_routing_use_new_routing_logic",
+                               GetParam().use_new_locality_routing ? "true" : "false"}});
+  // Add 3 hosts with weights {6, 3, 1}. Out of 10 refreshes with consecutive
+  // random value, 6 times the first host will be chosen, 3 times the second
+  // host will be chosen, and 1 time the third host will be chosen.
+  hostSet().healthy_hosts_ = {
+      makeTestHost(info_, "tcp://127.0.0.1:80", simTime(), 6),
```

Comment:
could you also test with non integer weights?

Reply:
i see that original weights are integer only, but perhaps testing with slow start config is worth to cover as well.